### DISCUSSION TITLE
How many bits of entropy does this use?

### DISCUSSION THREAD

**Discussion by GiverofMemory** [cathugger/mkp224o#95] [Q&A] [answered]
Love the tool!  How many bits of entropy does this use and where can I find that in the code?  As we know Profanity addresses have been hacked because that gen only used [32 bits of entropy](https://medium.com/coinmonks/the-profanity-address-hack-how-are-vanity-addresses-generated-cce40ba5ed39).

Thanks.

**Comment by scribblemaniac** [comment, marked as answer]
In normal usage, the seed size is defined by the SEED_LEN macro, which is 32 bytes/256 bits by default (the full private key size of ed25519). The seeds are randomly generated using libsodium, but the actual entropy of the seed will depend on the source of the random data (see https://libsodium.gitbook.io/doc/generating_random_data). Given that these functions are designed for cryptographic applications, they should have a reasonably high entropy.

If you use the passphrase option, your generated keys will only be as strong as the passphrase you use (and never a higher entropy than SEED_LEN). If a weak passphrase is used, it could be feasible to brute-force the key pair for a vanity address generated with mkp224o.

**Reply by GiverofMemory** [reply]
Perfect, thanks!  Very interesting about the passphrase option, I wasn't aware of that.
